monix-eval/shared/src/main/scala/monix/eval/Task.scala

```diff
@@ -808,7 +809,7 @@ sealed abstract class Task[+A] extends Serializable with TaskDeprecated.BinCompa
   @UnsafeBecauseImpure
   def runAsyncOptF(cb: Either[Throwable, A] => Unit)(implicit s: Scheduler, opts: Options): CancelToken[Task] = {
     val opts2 = opts.withSchedulerFeatures
-    Local.bindCurrentIf(opts2.localContextPropagation) {
+    Local.bindCurrentIf(opts2.localContextPropagation && opts2.localContextIsolateOnRun) {
```

Comment:
@Avasil Do you think that this is causing the issue where concurrent read/writes of `Local` stopped working if you have `disableLocalContextIsolateOnRun`?

Its the only thing that comes to mind, the rest of the functions just deal with running it to a `Future`?